stan/math/prim/prob/neg_binomial_2_lpmf.hpp

```diff
@@ -49,66 +48,62 @@ return_type_t<T_location, T_precision> neg_binomial_2_lpmf(
   size_t size_n_phi = max_size(n, phi);
   size_t max_size_seq_view = max_size(n, mu, phi);
 
-  VectorBuilder<true, T_partials_return, T_location> mu_val(size_mu);
-  for (size_t i = 0; i < size_mu; ++i) {
+  size_t len_ep = max_size(mu, phi);
+  size_t len_np = max_size(n, phi);
+
+  size_t len_mu = size(mu);
```

Comment:
This duplicates `size_mu`. By the way, note that for compatibility with C++17 compilers, all uses of `size()` should actually be `stan::math::size()` even if we are inside the `stan::math` namespace.